.golangci.yml

```diff
@@ -61,6 +61,7 @@ linters:
     - maligned          # readability trumps efficient struct packing
     - nlreturn          # generous whitespace violates house style
     - nosnakecase       # deprecated in https://github.com/golangci/golangci-lint/pull/3065
+    - perfsprint        # waiting for https://github.com/catenacyber/perfsprint/issues/21
```

Comment:
With this enabled,  you get errors like:

```
internal/cmd/dockerbuild/main.go:111:35: loopclosure: loop variable pluginGroup captured by func literal (govet)
                        return c.buildPluginGroup(ctx, pluginGroup, plugins)
                                                       ^
tests/plugins_test.go:132:44: loopclosure: loop variable toTest captured by func literal (govet)
                        if imageOverrides, ok := imageOverrides[toTest.Name]; ok {
                                                                ^
tests/plugins_test.go:136:28: loopclosure: loop variable toTest captured by func literal (govet)
                                testPluginWithImage(t, toTest, image)
                                                       ^
tests/plugins_test.go:138:7: loopclosure: loop variable toTest captured by func literal (govet)
                        if toTest.Name == "buf.build/grpc-ecosystem/gateway" && semver.Compare(toTest.PluginVersion, "v2.16.0") >= 0 {
                           ^
tests/plugins_test.go:128:2: Range statement for test TestGeneration does not reinitialise the variable toTest (paralleltest)
        for _, toTest := range plugins {
        ^
cmd/download-plugins/main.go:52:18: fmt.Sprintf can be replaced with string addition (perfsprint)
                releaseSince = fmt.Sprintf("%s.1", time.Now().UTC().Add(-since).Format("20060102"))
                               ^
internal/cmd/fetcher/main.go:136:50: fmt.Sprintf can be replaced with string addition (perfsprint)
        cmd := exec.CommandContext(ctx, "make", "test", fmt.Sprintf("PLUGINS=%s", strings.Join(pluginsEnv, ","))) //nolint:gosec
                                                        ^
internal/docker/build.go:31:3: fmt.Sprintf can be replaced with string addition (perfsprint)
                fmt.Sprintf("build.buf.plugins.config.owner=%s", identity.Owner()),
                ^
```

These might be good optimization for performance critical code but IMO they impact readability. There doesn't appear to be a setting to disable this type of lint failure - an upstream ticket https://github.com/catenacyber/perfsprint/issues/21 tracks it.

Reply:
Not sure I see the full value of this linter 🤷